proposals/0000-swift-turbo-modules.md

```diff
@@ -0,0 +1,185 @@
+---
+title: Swift Turbo Modules
+author:
+- Oskar Kwasniewski 
+- Riccardo Cipolleschi
+date: 12-08-2024 
+---
+
+# RFC0000: Swift Turbo Modules
+
+## Summary
+
+This RFC aims to allow developers to write Turbo Modules using Swift. This will allow the usage of more modern language making maintenance of native modules easier and more accessible. 
+
+## Motivation
+
+The primary motivations for introducing Swift Turbo Modules are:
+- Enhance developer experience for iOS developers working with React Native
+- Allow the use of more modern language
+- Make barrier to entry lower
+
+
+## Detailed design
+
+One of the problem why we can't adopt Swift in TurboModules  is the contamination of C++ ending up in user-space.
+
+### Current Situation
+The interfaces we generate for an Objective-C turbomodules have this shape:
+```objc
+@protocol NativeMyTurboModuleSpec <RCTBridgeModule, RCTTurboModule>
+@end
+```
+
+Where the `RCTTurboModule` protocol requires the conforming object to implement a method with this signature:
+```objc
+ - (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:
+       (const facebook::react::ObjCTurboModule::InitParams &)params;
+```
+
+As you can see, the signature of this method contains two types from C++:
+
+* `std::shared_ptr<facebook::react::TurboModule>`
+* `facebook::react::ObjCTurboModule::InitParams`
+
+### Solution
+
+The idea is to wrap the `getTurboModule` invocation in a `TurboModuleWrapper` object.
```

Comment:
```suggestion
The idea is to wrap the `getTurboModule` invocation in a `ModuleProvider` object.
```
or also

I think that Wrapper is not the right term for this pattern. we are calling a factory method to provide a new object, we are not wrapping a method call to adapt its interface or to decorate it with some other functionalities.